com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStep.java

```diff
@@ -47,23 +55,33 @@ protected StepPhase executeStep(ExecutionWrapper context) {
 
     private StepPhase attemptToExecuteStep(ExecutionWrapper execution) {
         CloudApplication app = StepsUtil.getApp(execution.getContext());
-        ApplicationStartupState currentState = computeCurrentState(execution, app);
+        CloudControllerClient client = execution.getControllerClient();
+        CloudApplication retrievedCloudApplication = client.getApplication(app.getName());
```

Comment:
The presence of two CloudApplication objects in the same method is still confusing IMO. Replace line 57 with:
```
// The rest of the fields in the object are null, so we need to retrieve them from the controller.
String appName = StepsUtil.getApp(execution.getContext()).getName();
```
Also, rename `retrievedCloudApplication` to `app` afterwards.